Summary Table/Summary Table Dev Flextable.Rmd

```diff
@@ -0,0 +1,398 @@
+library(gridExtra)
+library(grid)
+library(kableExtra)
+library(dplyr)
+library(flextable)
+library(officer)
+library(officedown)
+
+# detach(package:tidylog, unload = TRUE)
+```
+
+
+```{r echo=FALSE}
+# Set locality name object (with automatic wrapping)
+loc_name <- paste0(LOCALITY, "\nLocality") %>%
+  strwrap(width = case_when(n_loc < 4 ~ 20, n_loc == 4 ~ 15, n_loc > 4 ~ 8), simplify = FALSE)
+loc_name <- sapply(loc_name, paste, collapse = "\n")
+
+# Set other localities df
+otherloc_names <- paste0(other_locs$hscp_locality, "\nLocality") %>%
+  strwrap(width = case_when(n_loc < 4 ~ 20, n_loc == 4 ~ 15, n_loc > 4 ~ 8), simplify = FALSE)
+otherloc_names <- sapply(otherloc_names, paste, collapse = "\n")
+otherloc_names <- data.frame(loc_number = 2:n_loc, otherloc_names) %>%
+  pivot_wider(names_from = loc_number, values_from = otherloc_names)
+
+# Set HSCP name object
+hscp_name <- paste0(HSCP, "\nHSCP") %>%
+  strwrap(width = if_else(n_loc < 3, 30, 12), simplify = FALSE)
+hscp_name <- sapply(hscp_name, paste, collapse = "\n")
+
+summary_table <- data.frame(
+  Indicators = "Indicators",
+  Data_Type = "Data Type",
+  Time_Period = "Time Period",
+  Locality = loc_name,
+  Other_Localities = otherloc_names,
+  HSCP = hscp_name,
+  Scotland = "Scotland"
+)
+
+```
+
+## Demographics Summary Table
+
+
+```{r echo=FALSE}
+
+indicators <- c(
+  "Total population",
+  "Gender ratio male to female",
+  "Population over 65",
+  "Population in least deprived SIMD quintile",
+  "Population in most deprived SIMD quintile"
+)
+# List what data type they are
+datatype <- c("count", "ratio", "%", "%", "%")
+# Enter the time periods available
+time <- c(pop_max_year, pop_max_year, pop_max_year, "2020", "2020")
+
+loc_values <- c(total_population, paste0("1:", gender_ratio), over65, perc_top_quintile, perc_bottom_quintile)
+other_loc_values <- rbind(other_locs_total_pop, other_locs_gender_ratio, other_locs_over65, other_locs_simd)
+hscp_values <- c(hscp_total_pop, hscp_gender_ratio, hscp_over65, hscp_simd_top, hscp_simd_bottom)
+scot_values <- c(scot_total_pop, scot_gender_ratio, scot_over65, 20, 20)
+
+demographic_values <-data.frame(indicators, datatype, time, loc_values, other_loc_values,
+                          hscp_values, scot_values)
+
+headings <- as.character(unlist(summary_table[1, ]))
+
+colnames(demographic_values) <- headings
+
+demographics <- flextable(demographic_values) %>%
+  bold(part = "header") %>%
+  bold(j = 1, bold = TRUE, part = "body") %>%
+  fontsize(part = "header", size = 12) %>%
+  font(part = "body", fontname = "Arial") |>
+  font(part = "header", fontname = "Arial") |> 
+  color(part = "header", color = "white") %>%
+  bg(part = "header", bg = "#3F3685") %>%
+  bg(j = 1:3, part = "body", bg = "grey90") |>
+ # width(j = 1, width = "22em") |> 
+  border(border = fp_border(color = "white", width = 1), part = "body") |> 
+  set_table_properties(layout = "autofit")#|> 
+ # autofit()  # Automatically adjust column widths
```

Comment:
I would prefer to remove code, rather than comment it out if it's not needed. If it is needed then add (another) comment noting why.
```suggestion
  border(border = fp_border(color = "white", width = 1), part = "body") |> 
  set_table_properties(layout = "autofit")
```